### ISSUE TITLE
Verify data in CST cluster after the migration

### ISSUE THREAD

**Issue by NGA-TRAN** [influxdata/influxdb_iox#8616]
Since we found some [catalog data `sort_key_ids` was not fully filled](https://github.com/influxdata/influxdb_iox/issues/8570) and fixed them in Serverless after the [migration PR](https://github.com/influxdata/influxdb_iox/pull/8491) (its [ticket](https://github.com/influxdata/influxdb_iox/issues/8467)), we want to see how things work in CST clusters.

If all data of CST were migrated well, we will declare migration successful and move forward. If not, we will investigate why.

Note that the migration only affects EXISTING data. If somehow the auto-script migration is not fully covered all migration work and can be fixed manually, we will go ahead with that.

**Comment by NGA-TRAN** [comment]
Here are the clusters I am looking into 

```
tubectl list clusters
+--------------------------------------+-------------------+--------------------------------------+----------------------------+--------------+
| Account ID                           | Account Nickname  | Cluster ID                           | Cluster Nickname           | Region       |
+=============================================================================================================================================+
| 1276718b-f517-4d35-9db1-809ec610eeec | BTC-AG            | a2bca010-2c96-421c-8fec-8a48160b4cbf | BTC-AG-POC                 | eu-central-1 |
|--------------------------------------+-------------------+--------------------------------------+----------------------------+--------------|
| 476c4756-7a06-4907-a082-fd909b2693e8 | Hazeltree         | 4d961f9c-12ac-4e36-a22d-9935ed0970b5 | <missing cluster nickname> | us-east-1    |
|--------------------------------------+-------------------+--------------------------------------+----------------------------+--------------|
| 7e3e3e08-8af3-49fe-a5b4-13c44ff4ba8f | InfluxData        | b0c7cce5-8dbc-428e-98c6-7f996fb96467 | <missing cluster nickname> | us-east-1    |
|--------------------------------------+-------------------+--------------------------------------+----------------------------+--------------|
| 9c310174-9883-4883-a0c3-8dc07d32c308 | Senec             | 77a0ed26-dca9-417d-844a-d1359305906d | <missing cluster nickname> | eu-central-1 |
|--------------------------------------+-------------------+--------------------------------------+----------------------------+--------------|
| 9c310174-9883-4883-a0c3-8dc07d32c308 | SENEC             | 84984b87-7c2b-4d60-8f1a-964a1c1cda4f | SENEC-PROD                 | eu-central-1 |
|--------------------------------------+-------------------+--------------------------------------+----------------------------+--------------|
| a6597aa3-ede6-4f3f-a837-f538674b07c2 | InfluxData        | a2024295-b00e-49a3-8d30-5c06a1ec1061 | Support                    | us-west-2    |
|--------------------------------------+-------------------+--------------------------------------+----------------------------+--------------|
| b67bd8da-8de3-450c-b3a9-ddce91f30f18 | ASTSpaceMobile    | efc8c529-c0d1-4b2a-b52b-60db9c32262a | <missing cluster nickname> | us-east-1    |
|--------------------------------------+-------------------+--------------------------------------+----------------------------+--------------|
| cb0aba3c-3cb5-4f8b-848c-59db9810d6b9 | SmartPower        | 65741c4e-7652-4e57-82ea-24f263b67d0c | Production                 | eu-central-1 |
|--------------------------------------+-------------------+--------------------------------------+----------------------------+--------------|
| d25164e4-40c5-48e6-8cbe-650c4205ca35 | InfluxData        | f4b53153-32e2-4774-8e2a-74d5f0264403 | PerformanceD               | us-east-1    |
|--------------------------------------+-------------------+--------------------------------------+----------------------------+--------------|
| dff0a990-4e3c-4c0e-a736-3d97a4c75e94 | InfluxData        | bc576735-3428-4e8b-ae4b-d74732eb59db | PerformanceB               | us-east-2    |
|--------------------------------------+-------------------+--------------------------------------+----------------------------+--------------|
| dff0a990-4e3c-4c0e-a736-3d97a4c75e94 | InfluxData        | c82a1cf7-916c-4e5d-abe5-19fb5407f700 | PerformanceC               | us-east-2    |
|--------------------------------------+-------------------+--------------------------------------+----------------------------+--------------|
| e1d848cd-a024-41c2-bcef-e6b55c53224f | VerticalAerospace | e10faa5c-deef-45b7-9364-38e922063341 | <missing cluster nickname> | eu-west-2    |
|--------------------------------------+-------------------+--------------------------------------+----------------------------+--------------|
| e1ec0974-cc99-453b-b748-32b4c48a977c | Xylem             | f09cc434-4bee-42bb-9423-f132f9313bca | Xylem-POC                  | us-east-1    |
|--------------------------------------+-------------------+--------------------------------------+----------------------------+--------------|
| e64f16f2-77ce-46be-aea3-0addb6a4ca5d | Verint            | e1791e44-949e-4352-b495-74c224e73125 | <missing cluster nickname> | us-east-1    |
|--------------------------------------+-------------------+--------------------------------------+----------------------------+--------------|
| e6b6562c-828c-4597-b568-353fe63143b9 | MODE-Inc          | cfb40864-4f0d-4ad0-aec0-062abd03cc71 | mode-inc-poc               | us-west-2    |
|--------------------------------------+-------------------+--------------------------------------+----------------------------+--------------|
| dff0a990-4e3c-4c0e-a736-3d97a4c75e94 | InfluxData        | b2649cea-e5d7-11ed-a05b-0242ac120003 | PerformanceA               | us-east-1    |
+--------------------------------------+-------------------+--------------------------------------+----------------------------+--------------+
```

**Comment by NGA-TRAN** [comment]
**BTC-AG**

```SQL
scripts/psql-catalog-rds --management-cluster prod --account-id 1276718b-f517-4d35-9db1-809ec610eeec --cluster-id a2bca010-2c96-421c-8fec-8a48160b4cbf

select id, partition_key, sort_key_ids, sort_key from partition where cardinality(sort_key_ids) != cardinality(sort_key);
 id | partition_key | sort_key_ids | sort_key 
----+---------------+--------------+----------
(0 rows)

select id, sort_key, sort_key_ids from partition limit 10;
 id |                      sort_key                      |   sort_key_ids   
----+----------------------------------------------------+------------------
 10 | {IsVirtual,MeasurementTask,ObisCode,SerialNo,time} | {24,25,26,29,23}
 11 | {IsVirtual,ObisCode,MeasurementTask,SerialNo,time} | {12,14,13,17,11}
 14 | {IsVirtual,ObisCode,MeasurementTask,SerialNo,time} | {37,39,38,42,36}
 13 | {IsVirtual,ObisCode,MeasurementTask,SerialNo,time} | {2,4,3,6,1}
  3 | {IsVirtual,MeasurementTask,ObisCode,SerialNo,time} | {12,13,14,17,11}
  4 | {IsVirtual,MeasurementTask,ObisCode,SerialNo,time} | {24,25,26,29,23}
  1 | {IsVirtual,MeasurementTask,ObisCode,SerialNo,time} | {2,3,4,6,1}
  5 | {IsVirtual,MeasurementTask,ObisCode,SerialNo,time} | {37,38,39,42,36}
(8 rows)
```

**Hazeltree**

```SQL
scripts/psql-catalog-rds --management-cluster prod --account-id 476c4756-7a06-4907-a082-fd909b2693e8 --cluster-id 4d961f9c-12ac-4e36-a22d-9935ed0970b5

select id, partition_key, sort_key_ids, sort_key from partition where cardinality(sort_key_ids) != cardinality(sort_key);
 id | partition_key | sort_key_ids | sort_key 
----+---------------+--------------+----------
(0 rows)

select id, sort_key, sort_key_ids from partition limit 10;
 id |                                                    sort_key                                                    |                   sort_key_ids                    
----+----------------------------------------------------------------------------------------------------------------+---------------------------------------------------
  1 | {environment,host,instance,objectname,source,time}                                                             | {3,4,5,6,11,1}
  3 | {CompanyCode,Environment,GridMeta,ParentGridMeta,ViewMeta,environment,host,mtype,unit,unit_dur,unit_rate,time} | {75,79,80,93,104,78,81,87,101,102,103,74}
  4 | {CompanyCode,Environment,GridMeta,ParentGridMeta,ViewMeta,environment,host,mtype,unit,unit_dur,unit_rate,time} | {137,141,142,155,166,140,143,149,163,164,165,136}
  5 | {CompanyCode,Environment,GridMeta,ParentGridMeta,ViewMeta,environment,host,mtype,unit,unit_dur,unit_rate,time} | {44,48,49,62,73,47,50,56,70,71,72,43}
  7 | {CompanyCode,Environment,GridMeta,ParentGridMeta,ViewMeta,environment,host,mtype,unit,unit_dur,unit_rate,time} | {168,172,173,186,197,171,174,180,194,195,196,167}
  9 | {CompanyCode,Environment,GridMeta,ParentGridMeta,ViewMeta,environment,host,mtype,unit,unit_dur,unit_rate,time} | {199,203,204,217,228,202,205,211,225,226,227,198}
 10 | {environment,host,mtype,unit,time}                                                                             | {261,262,263,264,260}
 11 | {CompanyCode,Environment,GridMeta,ParentGridMeta,ViewMeta,environment,host,mtype,unit,unit_dur,unit_rate,time} | {298,302,303,316,327,301,304,310,324,325,326,297}
 13 | {CompanyCode,Environment,GridMeta,ParentGridMeta,ViewMeta,environment,host,mtype,unit,unit_dur,unit_rate,time} | {329,333,334,347,358,332,335,341,355,356,357,328}
 74 | {CompanyCode,Environment,ParentGridMeta,environment,host,mtype,unit,unit_dur,unit_rate,GridMeta,ViewMeta,time} | {428,432,446,431,434,440,454,455,456,433,457,427}
(10 rows)
```

**InfluxData1**

```
scripts/psql-catalog-rds --management-cluster prod --account-id 7e3e3e08-8af3-49fe-a5b4-13c44ff4ba8f --cluster-id b0c7cce5-8dbc-428e-98c6-7f996fb96467

 id | partition_key | sort_key_ids | sort_key 
----+---------------+--------------+----------
(0 rows)

select id, sort_key, sort_key_ids from partition limit 10;
  id   |                              sort_key                               |             sort_key_ids             
-------+---------------------------------------------------------------------+--------------------------------------
 24179 | {cluster_id,datacenter,host,node,url,db,time}                       | {1084,1085,1087,1088,1091,1086,1081}
  4558 | {charge_type,host,payment_method,tariff_code,tariff_direction,time} | {1307,1310,1311,1312,1465,1306}
  7844 | {charge_type,host,payment_method,tariff_code,tariff_direction,time} | {1307,1310,1311,1312,1465,1306}
  7355 | {charge_type,host,payment_method,tariff_code,tariff_direction,time} | {1307,1310,1311,1312,1465,1306}
  7664 | {charge_type,host,payment_method,tariff_code,tariff_direction,time} | {1307,1310,1311,1312,1465,1306}
  8415 | {charge_type,host,payment_method,tariff_code,tariff_direction,time} | {1307,1310,1311,1312,1465,1306}
  7300 | {charge_type,host,payment_method,tariff_code,tariff_direction,time} | {1307,1310,1311,1312,1465,1306}
  8510 | {charge_type,host,payment_method,tariff_code,tariff_direction,time} | {1307,1310,1311,1312,1465,1306}
  6953 | {charge_type,host,payment_method,tariff_code,tariff_direction,time} | {1307,1310,1311,1312,1465,1306}
  4261 | {charge_type,host,tariff_code,payment_method,tariff_direction,time} | {1307,1310,1312,1311,1465,1306}
(10 rows)
```

**Comment by NGA-TRAN** [comment]
Close as the verification is done. @crepererum and I will continue the migration work in https://github.com/influxdata/influxdb_iox/issues/8467

**Comment by crepererum** [comment]
Regarding `PerformanceB`:

```
iox_catalog=> select * from _sqlx_migrations ;
    version     |                  description                   |         installed_on          | success |                                              checksum                                              | execution_time
----------------+------------------------------------------------+-------------------------------+---------+----------------------------------------------------------------------------------------------------+----------------
 20210217134322 | create schema                                  | 2023-05-19 15:33:28.520941+00 | t       | \xb88c635e27f8b9ba8547b24efcb081429a8f3e85b70f35916e1900dffc4e6a77eed8a02acc7c72526dd7d50166b63fbd |        5188367
 20211229171744 | initial schema                                 | 2023-05-19 15:33:28.530284+00 | t       | \x6a17f931cef0cc7a3fe3d1ca018a71ac89c34df0fbf4548920f6f86abd3d63448ef807114b96387c1d51c0e335afbb57 |      206361461
 20220221145409 | routing limits                                 | 2023-05-19 15:33:28.740046+00 | t       | \xe9e65e0b697c4680a9da906d558b8428896118412d961778370a6c6ce5eaaadba6297d8fad5b9478977c88cce34835a1 |        6869210
 20220223192306 | add row count to parquet file                  | 2023-05-19 15:33:28.750115+00 | t       | \x4a8c60a87d5e8caee26eb56d066c2a50b9b7582c98fcbdfc6c1c89ed70e307926605bd35a6abb91421a4beb2fa72fd4b |        7801880
 20220224080910 | add file size bytes                            | 2023-05-19 15:33:28.760967+00 | t       | \x8d8355486e384bd909e0d79e030d44218a1f18e7893af904fd3f291a522d2a81e8bb18dc89305d25b2c9729c0105ea4a |        5997036
 20220224082147 | add parquet metadata                           | 2023-05-19 15:33:28.770042+00 | t       | \x7644f8ecb292608f30b92a9dc4ab92eba3cb9cd8f4009d6e98b505f00a66ac70a687ab3ef0abcdc0b1d97d4f7259f1b4 |        9964846
 20220307201050 | add parquet level and created at               | 2023-05-19 15:33:28.783163+00 | t       | \x5f72caaa2ad1b85a01c7b721b035a7e99a0b167346135a7c6894dec4ef5477e7b6004273ac43f9f14e93bc7e41cf9d67 |        8250586
 20220308120917 | ingester-missing-indexes                       | 2023-05-19 15:33:28.794478+00 | t       | \xecea4f58da78df90d6441931176b21037e7923b7eb393a84119f5d5e0c060a7230ceb3590b645c502a3b484213c37ed4 |       12826939
 20220323150705 | change to delete to timestamp                  | 2023-05-19 15:33:28.810418+00 | t       | \x5287ad1996d2b3aeb10b7e243c177c093e141f578d033252859c0d61d770350102e205ed81488b885f4153ddc06482a3 |       30444681
 20220324152729 | add namespace id to parquet file               | 2023-05-19 15:33:28.845682+00 | t       | \x2092adf7071ec79e52ea1d4b5535e82232cf55482da81cf4713ec4948cd659d8eb3e2e383927cbcab971bad89ab7cc0c |       10789898
 20220328150925 | add index to to delete                         | 2023-05-19 15:33:28.859431+00 | t       | \x62a6188068e31e2dd0b087cd32270c3914faa3480d1511e64cdd3384a1ee00adbc4a8a9e2d3e63ad8600526396e59eb2 |        9548834
 20220331222147 | compactor-indexes                              | 2023-05-19 15:33:28.872263+00 | t       | \xc0a664be63831d563191d9905f63102962865783ea30e29789a13e9d30376153c74c4f906e2121da9a106a29d923595a |       14733844
 20220401174403 | add sort key to partition                      | 2023-05-19 15:33:28.890004+00 | t       | \x6b716bbbc065ba1e9cc6982c47d0b8ad508760f35ed10c2c813de9697261388b765ddb29e79048f086add8ddafd8de04 |        5402513
 20220404120116 | column name-table id-index                     | 2023-05-19 15:33:28.898453+00 | t       | \x3258d07ba3f856af152edfc1590fdc855a644821d476b037dbd2966fa32d2b2edaceedb1ac874a5e9c4d5685098223e1 |        9850840
 20220420142224 | ids bigint                                     | 2023-05-19 15:33:28.911166+00 | t       | \x018736d153cca9bcba25c5bec8ffe2a0e9b0b9d9ce687cfecdacbc05f4cb3505fc2f60319194fd161cc2ce7f166ae9a2 |      402650614
 20220610102325 | add sort key array column                      | 2023-05-19 15:33:29.329553+00 | t       | \x0cbcaf3d04d9d392c7a589153fbdb27fec239955a834b11729ab040020b431668b095034f3d5599432bb5dab183a8c43 |        7551830
 20220610131325 | rename sort key arr                            | 2023-05-19 15:33:29.340161+00 | t       | \xfaa7a14d436b808912261d0b6b667976daadd2bae36bdea9e81462c28f8046740dcaecbae4b077806e594a5aa06a7f43 |        8823922
 20220620160136 | parquet file columns                           | 2023-05-19 15:33:29.351968+00 | t       | \x89761be112fc02d2d6a823b7fb011aaa94aedd4714c236c083e5fdccbdd13666f802e63bbab932b895c469ced2717c56 |        5280706
 20220623143430 | non null parquet file columns                  | 2023-05-19 15:33:29.360211+00 | t       | \x1f8bedffc127c27b6b83c97eb594dd1d08b3845c5afa039d409ec9cb3147d49900afd626187ef8db2b014ba13d3fb40f |        5418884
 20220624104030 | remove column sort key old                     | 2023-05-19 15:33:29.368654+00 | t       | \xcbdbf461941d03bc6c760a91131f01639e6ada4c2f177366f5a757658190f5ff864fd7352feba6810c157fbb344e0476 |        5647258
 20220627092839 | remove parquet metadata                        | 2023-05-19 15:33:29.377271+00 | t       | \x3c25a15ca767ff6ebdb21d07603725bd48e6a51c2939c5c1fc3680ed5b9acebae32eb90a564dd44dba6e28e1f8789eba |        7024420
 20220628085556 | column set via IDs                             | 2023-05-19 15:33:29.387376+00 | t       | \x7742751f8feb4ff6535e4d20a65666b308e810c93885bdd9660698c2ffd43094b1d8bd1eb06fdee1c42255eefa4e63a5 |       35875037
 20220721195556 | drop min sequence number                       | 2023-05-19 15:33:29.426838+00 | t       | \x38bb5c9ba4078effb9dfbe3611ef06d931054d8e6edbce4fd3d7e739c85f24cbb893a44aebcff76eddd21832f024f80d |        5956351
 20220819154515 | sequencer to shard                             | 2023-05-19 15:33:29.436245+00 | t       | \xf7abd1f1b6a4f6d9ea78bc28ac1dcc2a0b8829ae0b8db9571733f495abfdbffe929aa1f368f235ec153c4bd4415358d0 |       10754633
 20220902145700 | compactor index                                | 2023-05-19 15:33:29.450063+00 | t       | \xb4869e0891d83e61b6a1520657d97c60afe9c4cf0b799fd319320b7dbe5dd908f5aa529cf257c863abf41788400d2592 |       10183881
 20220909140048 | create skipped compactions                     | 2023-05-19 15:33:29.463219+00 | t       | \xf12d5b742e1e326392ee46557ed762b59fb6c645332551fd154911549f12700a32ad9844e92e7b61e3ad8f9529c4c4d7 |       23342028
 20220913162700 | parquetfile filesize trigger                   | 2023-05-19 15:33:29.489593+00 | t       | \xbbdac9dd1261cae0140ac9529fa4a3350ac21771bdc96a53d23bf219ba75edb51576db224de87db51e40a558eb35d528 |       45604657
 20220915085930 | per-partition-persist-offset                   | 2023-05-19 15:33:29.53843+00  | t       | \xf3ac2849bbda91f0401e72361ec685f5aeb1620863690b34e15c89fb0b204fd4f39bb7b9812bd35f8d60a0f21cebb4e3 |        6658496
 20220915155941 | nullable-partition-persist-offset              | 2023-05-19 15:33:29.548148+00 | t       | \xc658376480104322de7610eb15d2e17b60d33df07053dd9fa9458612bf18b16cbc650c98b30c7d3643e6c6837bbba5a8 |        5925868
 20220926182450 | add cols to skipped compactions                | 2023-05-19 15:33:29.557049+00 | t       | \x77d9489016e80bf3ff36946184d4c6276d5c6fdf4c0de2411e4e5162b6c94799e3af43a336e7377a8a0085a6e842fd59 |        5500338
 20221014122742 | lower-default-per-table-column-limit           | 2023-05-19 15:33:29.565517+00 | t       | \xea6d4cb7b654ad780fbf586c4476823db2f3b027f56c366edea1f0fbc308acd03fc0cae4a2cca9ebf1d707311cafe7b3 |        8150631
 20221101170001 | add cols for retention policy                  | 2023-05-19 15:33:29.576768+00 | t       | \xc1960c77de8ff3d335614916e7f5599861e1a7f9998123d4421f0515ffc5437abba8b5a89dcf082afd1cc7b55f40e768 |        6719633
 20221111133939 | drop retention duration                        | 2023-05-19 15:33:29.586425+00 | t       | \xc6069822e251c12dbac8dc9451d08daaf0bfbf06c8f9b60af41673d24305494374327289d8a4f78936af7d407d10b0f4 |        5324569
 20221118135602 | add col for compaction first file in partition | 2023-05-19 15:33:29.594935+00 | t       | \xce7347155760c7473a9605071cd5c58f30ec955c72e2cd609777e25116d872277524b2b1882b42ff85fe44f52d5fb849 |        5412562
 20221215145057 | namespace delete cascade                       | 2023-05-19 15:33:29.603377+00 | t       | \xcbb7a407338aea53acaf791e49937b3352e082514b1644cd0d23ad7511077b7f958542658032db01a75944e2a6f6c625 |       22909887
 20230104114545 | create indexes without shard                   | 2023-05-19 15:33:29.62933+00  | t       | \x4872f119a962f710b5b9f42f7b946a54346a3f150a48f65b84de62ecb618fd4cf99e14862e75e3dc02156907371ef4c5 |        9732622
 20230105120822 | trigger update partition                       | 2023-05-19 15:33:29.641965+00 | t       | \x760e842dec9e56cf82a8178e370a46c5bdfe02656ccaaec62b6b1a6c06e1f11e13e1720f34f36eb85adf1bb5b883b0d5 |        8952016
 20230106145820 | modify trigger                                 | 2023-05-19 15:33:29.654053+00 | t       | \x6f7a1f9b4c13ef5752a95b77f4924be0c7ccf4a01344f2170d98639d63620d53b4b231d3c300cdbeafbd4ae5bdb87555 |        6105243
 20230113075522 | modify trigger new file at                     | 2023-05-19 15:33:29.663207+00 | t       | \xca39237c6bee5ef9e43984c3f367ab852813bf38bf5c8c22c1dbd89b4f003849a8580767bc60ca5197dd7272bffab56e |        4718292
 20230125103737 | create max l0 crrated at                       | 2023-05-19 15:33:29.670845+00 | t       | \xd97e4c6991eabf4f65a994d6fefe8a510ea9c216a72c02c073a59a2aa6022b1448bc834cde567983af571ef595dfcccb |        6302496
 20230207103859 | namespace-soft-delete-column                   | 2023-05-19 15:33:29.6801+00   | t       | \x874322235c18d1b45f61a7c22857a481dead3a17dffec246a7edf1e11bd9cc40e85cd345c41e69ebf54c088405992f03 |        9951062
 20230501173720 | add partition templates                        | 2023-05-25 11:59:51.325404+00 | t       | \xd28d3696d3fc1cd682a8e02fa4f019c133e70a2fd60489d6ad725e116ddcaa391cf8c9ef91aaea79d76942fee46e52c6 |       25795477
 20230522145017 | remove processed tombstones table              | 2023-05-25 11:59:51.355413+00 | t       | \x7ecb895ede9601fb9591781e295a44ad08255b8eee3d527409f557ed8dff46f7891d416b3abe904582413e364df26c6a |       32428517
 20230524095337 | remove unused parquet file indices             | 2023-05-25 11:59:51.39134+00  | t       | \xb1b5c2856739fb1f1f3cace4eb42098fcde5ca785a2d47e246ce8603b0df4840b14924fa018e0d17d86ebe44c5c057a9 |       16121904
 20230524151854 | add parquet file table and deleted index       | 2023-06-01 10:04:55.002214+00 | t       | \xddc52db62ed446e4a8fe30af7bced52724bcf93e6e6c6cac8cbd3783bf7312595cfc44dc6aa282db9ac58e4ecfb08268 |       58807632
 20230530132809 | add parquet file partition and deleted index   | 2023-06-01 10:04:55.063306+00 | t       | \x4b8295a25aa051620c8fe5fa64b914901a2f4af4d343d2735cde484aa018691e22d274aab84200ba2830fbe51833ab1a |       21350826
 20230601130730 | remove unused parquet file indices 2           | 2023-06-15 00:35:51.202917+00 | t       | \x1a4fc71243948a9c591bcfc93ff43621d575170a2fe2a858af41b4703f40fd3179674059d7ab783a183716905e5e7236 |       47796970
 20230417173102 | deterministic partition id                     | 2023-06-23 11:36:00.827568+00 | t       | \x3c2e3235ecfae18f021fd728261c698d0fd16ef0d49760a58ae4be9439bc273073dbd4a69fecc1f2e7f602ab8cde9e48 |      248343984
 20230707195110 | add created at index                           | 2023-07-11 09:16:40.498204+00 | t       | \x7a353a9a9876a691c6df91af2f2774bb0a43e5f86aa65ca470c16e413ef1741f4a75505ad4e56f1ecb206c2097aad90f |       58904348
 20230710141740 | partition hash id index                        | 2023-07-13 09:50:08.94394+00  | t       | \x2ee2416cc206254f5b8a5497a3cfc5bcb2146759416cd4cb6a83ae34d3e0141387eb733e61f224076d5af0e3c6016e7b |      129793579
 20230714154828 | make-kafka-columns-nullable                    | 2023-07-21 14:42:43.195818+00 | t       | \xa0f30a00e366a97d4faad88cae9f81f0bfa292c5e17bc8b7046598ed7b937841db86066cf745de14652af55d7a7fc1ed |       71751573
 20230720132400 | modify parquet file triggers                   | 2023-07-21 14:42:43.270403+00 | t       | \x54124874e57a7977f278bd8e7fb46abdb4c9e8faa67170056feb4966b5d15e5b24aad2d19c74e35bc8c9de235e197aa8 |       31424729
 20230726175943 | make parquet file partition id optional        | 2023-08-02 01:05:49.293033+00 | t       | \xa596c46838ed34cebace317a676140334b68ec1204fee02acfbcc0b9e12bd737545cb8c04f5008f555e579269652b71d |       36283785
 20230731143030 | add column sort key array ids                  | 2023-08-02 01:05:49.337928+00 | t       | \x43a2f131d1a26760d3a9b55dfee758ebfa9193f8498bb4cf75bd85409efaec72462641f06380ff621d545bf6c99f99d9 |        8723758
 20230814162311 | sort key ids add helper index                  | 2023-08-23 17:51:41.660873+00 | t       | \x4e2d4796117de5df48a4423d9dbe266c4a8967086b3f8814d2968ab76b2f7c90d374e7835e2e2b280b74733ed197ec70 |      157191783
 20230814162327 | sort key ids from names                        | 2023-08-23 17:51:41.887076+00 | t       | \xc51fd751c25d0d51a7a63fa940b9b56c597c760eca2c03fcb21d860666d19a3d0f0fb377e5844f3663563d637bdbdfdb |        2467745
 20230814162336 | sort key ids remove helper index               | 2023-08-23 17:51:41.897376+00 | t       | \x7e82c192ae103f891b0dfdf232041a646f20d5600b212efc7da0c2e1ad4ee12fce40e02963093e818925022b35e0624e |        2477733
 20230815091507 | sort key ids not null                          | 2023-08-23 17:51:41.906505+00 | t       | \x7c19141882fbc22a483b309bceda2c34c43a584940ca96c328452740715a5cc77f869d9a9e797ee023217386552cda7b |        2443527
(58 rows)
```

> This tells me the migration (aka filling sort_key_ids) seems not yet run on this CST clusters

Correct. The actual non-reverted migration (including the fix for empty keys) is in `20230822120632`..`20230822120825` (a block of 4 migrations).

**Comment by cannonpalms** [comment]
`PerformanceB` now has all migrations. It was frozen on an older build. No longer!
